Sources/Datadog/RUM/Instrumentation/RUMInstrumentation.swift

```diff
@@ -61,13 +39,15 @@ internal final class RUMInstrumentation: RUMCommandPublisher {
         configuration: FeaturesConfiguration.RUM.Instrumentation,
         dateProvider: DateProvider
     ) {
-        var viewsAutoInstrumentation: ViewsAutoInstrumentation?
+        viewsHandler = RUMViewsHandler(dateProvider: dateProvider)
+        var viewsAutoInstrumentation: UIViewControllerSwizzler?
         var userActionsAutoInstrumentation: UserActionsAutoInstrumentation?
         var longTasks: LongTaskObserver?
 
         do {
             if let predicate = configuration.uiKitRUMViewsPredicate {
-                viewsAutoInstrumentation = try ViewsAutoInstrumentation(predicate: predicate, dateProvider: dateProvider)
+                viewsHandler.predicate = predicate
```

Comment:
I think it was much safer when `predicate` was injected in initializer and its access was scoped to `private let` instead of `internal var`. This was guaranteeing state consistency in `RUMViewsHandler` and was easier to follow the logic. What's the reason to assign the `predicate` after init?

Reply:
The `viewsHandler` property is now non-optional, this makes the implementation of the `RUMInstrumentation.init` quite complex given conditions in `FeaturesConfiguration.RUM.Instrumentation` and throwable underlying init. I will see if I can improve that, I agree it should be best to keep the `predicate` immutable.

Reply:
I think just following could be safe enough and not that complex, no?
```swift
viewsHandler = RUMViewsHandler(
   dateProvider: dateProvider, 
   predicate: configuration.uiKitRUMViewsPredicate
)

do {
   if configuration.uiKitRUMViewsPredicate != nil {
      // UIKit auto instrumentation is enabled, so we need to install swizzler:
      viewsAutoInstrumentation = try UIViewControllerSwizzler(handler: viewsHandler)
   }
} ...
```

Reply:
Yep, that works :)